### PR TITLE
:tada: Destination Amazon SQS: New connector

### PR DESCRIPTION
**Pull request by sdairs**
## What
Adds an Amazon SQS Destination connector
Has Unit tests for write/check with mock AWS resources to simulate behaviour without needing to create a real SQS topic or IAM roles

Currently supports sending messages individually, rather than batch.

This has scope to improve by adding batched sends, configurable attributes, etc. 
There are some commented out code blocks where I have started to prototype these additional features, but the core is stable & tested such that it could be merged & these features completed/added later.

https://github.com/airbytehq/hacktoberfest-2021/issues/69

## How
*Describe the solution*

## Recommended reading order
1. `x.java`
2. `y.python`

## Pre-merge Checklist
Expand the relevant checklist and delete the others. 

<details><summary> <strong> New Connector </strong></summary>
<p>
   
#### Community member or Airbyter
   
- [x] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [x] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [x] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [x] Documentation updated 
    - [x] Connector's `README.md`
    - [x] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [x] `docs/SUMMARY.md`
    - [x] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [x] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [x] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] Connector added to connector index like described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
   
</p>
</details>

<details><summary> <strong> Updating a connector </strong></summary>
<p>
   
#### Community member or Airbyter
   
- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated 
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] Connector version bumped like described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</p>
</details>

<details><summary> <strong> Connector Generator </strong> </summary>
<p>
   
- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed.
</p>
</details>

### CODE REVIEW

**Comment by sdairs** [comment]
Unit test output

Check: 

- Tests with both a standard and FIFO queue.

Write: 

- Tests a Standard SQS queue with the full AirbyteMessageRecord's data as the Message body
- Tests a Standard SQS queue where the Message Body is user-supplied Key within the Record body.
- Tests a FIFO SQS queue with the full AirbyteMessageRecord's data as the Message body

```
==================================================================================================================== test session starts =====================================================================================================================
platform linux -- Python 3.9.7, pytest-6.2.5, py-1.10.0, pluggy-1.0.0 -- /usr/bin/python
cachedir: .pytest_cache
rootdir: /data/2/dev/airbyte/airbyte, configfile: pytest.ini
collected 2 items

unit_tests/unit_test.py::test_check {"type": "LOG", "log": {"level": "INFO", "message": "Found credentials in environment variables."}}
{"type": "LOG", "log": {"level": "DEBUG", "message": "Amazon SQS Destination Config Check - queue_url: https://eu-west-1.queue.amazonaws.com/123456789012/amazon-sqs-mock-queue"}}
{"type": "LOG", "log": {"level": "DEBUG", "message": "Amazon SQS Destination Config Check - region: eu-west-1"}}
{"type": "LOG", "log": {"level": "DEBUG", "message": "Amazon SQS Destination Config Check - access_key (ends with): I"}}
{"type": "LOG", "log": {"level": "DEBUG", "message": "Amazon SQS Destination Config Check - secret_key (ends with): 8"}}
{"type": "LOG", "log": {"level": "DEBUG", "message": "Amazon SQS Destination Config Check - Starting connection test ---"}}
{"type": "LOG", "log": {"level": "DEBUG", "message": "Amazon SQS Destination Config Check - Connection test successful ---"}}
{"type": "LOG", "log": {"level": "DEBUG", "message": "Amazon SQS Destination Config Check - queue_url: https://eu-west-1.queue.amazonaws.com/123456789012/amazon-sqs-mock-queue.fifo"}}
{"type": "LOG", "log": {"level": "DEBUG", "message": "Amazon SQS Destination Config Check - region: eu-west-1"}}
{"type": "LOG", "log": {"level": "DEBUG", "message": "Amazon SQS Destination Config Check - access_key (ends with): I"}}
{"type": "LOG", "log": {"level": "DEBUG", "message": "Amazon SQS Destination Config Check - secret_key (ends with): 8"}}
{"type": "LOG", "log": {"level": "DEBUG", "message": "Amazon SQS Destination Config Check - Starting connection test ---"}}
{"type": "LOG", "log": {"level": "DEBUG", "message": "Amazon SQS Destination Config Check - Connection test successful ---"}}
PASSED
unit_tests/unit_test.py::test_write {"type": "LOG", "log": {"level": "INFO", "message": "Found credentials in environment variables."}}
## Starting standard queue test ##
Listening for messages.
Message received.
Received message matches for standard queue write.
## Starting body key queue test ##
Listening for messages.
Message received.
Received message matches for body key queue write.
## Starting FIFO queue test ##
Listening for messages.
Message received.
Received message matches for FIFO queue write.
PASSED

====================================================================================================================== warnings summary ======================================================================================================================
airbyte-integrations/connectors/destination-amazon-sqs/unit_tests/unit_test.py: 1918 warnings
  /home/ab/.local/lib/python3.9/site-packages/responses/__init__.py:442: DeprecationWarning: stream argument is deprecated. Use stream parameter in request directly
    warn(

-- Docs: https://docs.pytest.org/en/stable/warnings.html
============================================================================================================== 2 passed, 1918 warnings in 3.76s ==============================================================================================================
```

**Comment by harshithmullapudi** [comment]
Hey @sdairs thanks for the contribution. Will look into it soon

**Comment by sdairs** [comment]
Did you get a chance to review @harshithmullapudi ?

**Comment by marcosmarxm** [comment]
Hello @sdairs ! Sorry to not finished your contribution before the date stipulated in the contest. All contributions made before 15-November are eligible to receive the award. We're trying to review your contribution as soon as possible.

**Comment by sdairs** [comment]
@marcosmarxm No worries, there's no rush 😄

**Comment by harshithmullapudi** [comment]
Hey @sdairs sorry for the delay. Can you run `./gradlew format` and also merge master into your branch ?

File: airbyte-integrations/connectors/destination-amazon-sqs/destination_amazon_sqs/destination.py [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+    #     }
+    #     response = queue.send_messages(Entries=messages)
+    #     if 'Successful' in response:
+    #         for status in response['Successful']:
+    #             print("Message sent: " + status['MessageId'])
+    #     if 'Failed' in response:
+    #         for status in response['Failed']:
+    #             print("Message sent: " + status['MessageId'])
+
+    def write(
```
Comment (harshithmullapudi) [inline airbyte-integrations/connectors/destination-amazon-sqs/destination_amazon_sqs/destination.py:75]: can you add the documentation link to the SQS write API here ?

File: airbyte-integrations/connectors/destination-amazon-sqs/destination_amazon_sqs/destination.py [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+    def add_attributes_to_message(self, record, message):
+        attributes = {"airbyte_emitted_at": {"StringValue": str(record.emitted_at), "DataType": "String"}}
+        message["MessageAttributes"] = attributes
+        return message
+
+    def set_message_delay(self, message, message_delay):
+        message["DelaySeconds"] = message_delay
+        return message
+
+    def set_message_fifo_properties(self, message, message_group_id, use_content_dedupe=False):
```
Comment (harshithmullapudi) [inline airbyte-integrations/connectors/destination-amazon-sqs/destination_amazon_sqs/destination.py:49]: can you add the documentation link to have better picture on why we need these properties ?

**Review by harshithmullapudi — CHANGES_REQUESTED** [review]

File: airbyte-integrations/connectors/destination-amazon-sqs/destination_amazon_sqs/destination.py (hunk 2 above)
Comment (sdairs) [reply airbyte-integrations/connectors/destination-amazon-sqs/destination_amazon_sqs/destination.py:51]: Added documentation link and comment. These are required properties specifically when using FIFO queues

File: airbyte-integrations/connectors/destination-amazon-sqs/destination_amazon_sqs/destination.py (hunk 1 above)
Comment (sdairs) [reply airbyte-integrations/connectors/destination-amazon-sqs/destination_amazon_sqs/destination.py:80]: Added a link to the SQS SendMessage API doc

**Comment by sdairs** [comment]
@harshithmullapudi Thanks - Merged master, formatted & added docs links

**Comment by harshithmullapudi** [comment]
@sdairs thanks for the contribution again. 🔥

**Pull request merged**